src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -431,9 +431,13 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
         <SkeletonLayout
           ref={intersectionObserverRef}
           style={{
-            [globalVars.stickyVerticalTopOffset]: `${verticalOffsets.header}px`,
-            [globalVars.stickyVerticalBottomOffset]: `${placement.insetBlockEnd}px`,
             paddingBlockEnd: splitPanelOpen && splitPanelPosition === 'bottom' ? splitPanelReportedSize : '',
+            ...(hasToolbar
+              ? {
+                  [globalVars.stickyVerticalTopOffset]: `${verticalOffsets.header}px`,
+                  [globalVars.stickyVerticalBottomOffset]: `${placement.insetBlockEnd}px`,
+                }
+              : {}),
```

Comment:
> Why with a toolbar the sticky position is not needed anymore?

It's needed, but when the top offset is added for a nested app layout, it add this offset twice, because this calculation is based on the global header, which results to doubled offset

> it doesn't seem that removing this condition would have any effect on the sticky behavior.

This is what is happens when this condition is removed:
![image](https://github.com/user-attachments/assets/426040c8-6414-470c-af51-60aaac61795e)
And this is what it's present:
![image](https://github.com/user-attachments/assets/1fa9e101-9f56-49be-9d27-6136ee0876ac)

Could you share the exact steps you did when removing this condition did not change the behavior?

Reply:
Alright, I dug a bit deeper. The reason seems to be that `useAppLayoutToolbarEnabled` doesn't return `true` so what I'm getting in my dev env is not `<ToolbarAppLayout />` but `<RefreshedAppLayout />`:

https://github.com/cloudscape-design/components/blob/31cd27dc010abe8a73286baf449f499cc7d46360/src/app-layout/internal.tsx#L16

The reason is that `getGlobalFlag` here 👇  is `undefined`.

https://github.com/cloudscape-design/components/blob/31cd27dc010abe8a73286baf449f499cc7d46360/src/app-layout/utils/feature-flags.ts#L9